pallets/ddc-staking/src/lib.rs

```diff
@@ -737,9 +835,94 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Pay out all the stakers for a single era.
+		#[pallet::weight(100_000)]
+		pub fn payout_stakers(origin: OriginFor<T>, era: EraIndex) -> DispatchResult {
+			ensure_signed(origin)?;
+			let current_era = Self::current_era().ok_or(Error::<T>::DDCEraNotSet)?;
+
+			// Makes sure this era hasn't been paid out yet
+			ensure!(!Self::paid_eras(era), Error::<T>::DoubleSpendRewards);
+
+			// This should be adjusted based on the finality of validation
+			ensure!(current_era >= era + 2, Error::<T>::EraNotValidated);
+
+			PaidEras::<T>::insert(era, true);
+			Self::do_payout_stakers(era)
+		}
+
+		/// Set price per byte of the bucket traffic in smallest units of the currency.
+		///
+		/// The dispatch origin for this call must be _Root_.
+		#[pallet::weight(10_000)]
+		pub fn set_pricing(origin: OriginFor<T>, price_per_byte: u128) -> DispatchResult {
+			ensure_root(origin)?;
+			<Pricing<T>>::set(Some(price_per_byte));
+			Ok(())
+		}
 	}
 
 	impl<T: Config> Pallet<T> {
+		pub fn do_payout_stakers(era: EraIndex) -> DispatchResult {
+			// ToDo: check that validation is finalised for era
+
+			let era_reward_points: EraRewardPoints<T::AccountId> =
+				<ErasEdgesRewardPoints<T>>::get(&era);
+
+			let price_per_byte: u128 = match Self::pricing() {
+				Some(pricing) => pricing,
+				None => Err(Error::<T>::PricingNotSet)?,
+			};
+
+			// An account we withdraw the funds from and the amount of funds to withdraw.
+			let payout_source_account: T::AccountId =
+				T::StakersPayoutSource::get().into_account_truncating();
+
+			// Transfer a part of the budget to each CDN participant rewarded this era.
+			for (stash, points) in era_reward_points.clone().individual {
+				let reward: BalanceOf<T> = match (points as u128 * price_per_byte).try_into() {
```

Comment:
Looks weird that we multiply `reword_points` by `price_per_byte`

Reply:
At least `price_per_byte` exposes the meaning of the value it contains. It is not price per reward point. At the same time, reward points don't strictly represent the volume of traffic CDN participants delivered because a portion of it could be cut by the validator in case of penalty is applied to the reward (for a high failure rate for instance).

Of course, the method should be better described in docs and maybe this kind of business logic should go to the separate trait in future releases.